File: app/src/main/java/com/example/rickydanobantonare/smartrice/Pest.java
package com.example.rickydanobantonare.smartrice;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ImageButton;
import android.widget.ListView;

public class Pest extends AppCompatActivity {
    ListView list;
    String[] pestename = {"Army Worm","Golden Apple Snail","Green Leafhopper","Rice Black Bug", "Rice Ear Bug"};
    String[] desc= {"Pest","Pest","Pest","Pest","Pest"};
    Integer[] imgid={R.drawable.armyworm,R.drawable.goldenapplesnail,R.drawable.greenleafhopper,R.drawable.riceblackbug, R.drawable.riceearbug};
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_pest);

        list = (ListView) findViewById(R.id.pestitem);
        CustomListView customListview=new CustomListView(this, pestename,desc,imgid);
        list.setAdapter(customListview);

        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> adapterView, View view, int position, long l) {
                switch (position){
                    case 0:
                        Intent intent = new Intent();
                        intent.setClass(Pest.this, PestsDefinition.class);
                        startActivity(intent);
                        break;
                    case 1:
                        Intent intent1 = new Intent();
                        intent1.setClass(Pest.this, PestsDefinition.PestDefinition2.class);
                        startActivity(intent1);
                        break;
                    case 2:
                        Intent intent2 = new Intent();
                        intent2.setClass(Pest.this, PestsDefinition.PestDefinition3.class);
                        startActivity(intent2);
                        break;
                    case 3:
                        Intent intent3 = new Intent();
                        intent3.setClass(Pest.this, PestsDefinition.PestDefinition4.class);
                        startActivity(intent3);
                        break;
                    case 4:
                        Intent intent4 = new Intent();
                        intent4.setClass(Pest.this, PestsDefinition.PestDefinition5.class);
                        startActivity(intent4);
                        break;
                }
            }
        });


        ImageButton androidImageButton = (ImageButton) findViewById(R.id.imageButton3);
        androidImageButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                backActivity();
            }
        });

    }
    public void backActivity(){
        Intent intent = new Intent(this, MainActivity.class);
        startActivity(intent);
    }
    @Override
    public void onBackPressed(){
        Intent intent = new Intent(Pest.this, MainActivity.class );
        startActivity(intent);
    }
}

File: app/src/main/kotlin/io/github/projectbukirin/palaylab/Pest.kt
package io.github.projectbukirin.palaylab

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import android.view.View
import android.widget.AdapterView.OnItemClickListener
import android.widget.ImageButton
import android.widget.ListView
import androidx.appcompat.app.AppCompatActivity
import io.github.projectbukirin.palaylab.PestsDefinition.*

class Pest : AppCompatActivity() {
    var list: ListView? = null
    var pestename: Array<String?> = arrayOf("Army Worm", "Golden Apple Snail", "Green Leafhopper", "Rice Black Bug", "Rice Ear Bug")
    var desc: Array<String?> = arrayOf("Pest", "Pest", "Pest", "Pest", "Pest")
    var imgid: Array<Int?> = arrayOf(R.drawable.armyworm, R.drawable.goldenapplesnail, R.drawable.greenleafhopper, R.drawable.riceblackbug, R.drawable.riceearbug)
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_pest)

        setSupportActionBar(findViewById(R.id.toolbar_pests))
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.title = "Pests"

        list = findViewById<View>(R.id.pestitem) as ListView
        val customListview = CustomListView(this, pestename, desc, imgid)
        list!!.adapter = customListview
        list!!.onItemClickListener = OnItemClickListener { adapterView, view, position, l ->
            when (position) {
                0 -> {
                    val intent = Intent()
                    intent.setClass(this@Pest, PestsDefinition::class.java)
                    startActivity(intent)
                }
                1 -> {
                    val intent1 = Intent()
                    intent1.setClass(this@Pest, PestDefinition2::class.java)
                    startActivity(intent1)
                }
                2 -> {
                    val intent2 = Intent()
                    intent2.setClass(this@Pest, PestDefinition3::class.java)
                    startActivity(intent2)
                }
                3 -> {
                    val intent3 = Intent()
                    intent3.setClass(this@Pest, PestDefinition4::class.java)
                    startActivity(intent3)
                }
                4 -> {
                    val intent4 = Intent()
                    intent4.setClass(this@Pest, PestDefinition5::class.java)
                    startActivity(intent4)
                }
            }
        }
    }

    fun backActivity() {
        val intent = Intent(this, MainActivity::class.java)
        startActivity(intent)
    }

    override fun onOptionsItemSelected(item: MenuItem) = when (item.itemId) {
        R.id.action_settings -> {
            val intent = Intent(this, About::class.java)
            startActivity(intent)
            true
        }

        else -> {
            super.onOptionsItemSelected(item)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.settings_actionbar, menu)
        return true
    }


    override fun onSupportNavigateUp(): Boolean {
        finish()
        return true
    }
}